src/lib.rs

```diff
@@ -823,6 +833,39 @@ impl Config {
         return dst;
     }
 
+    fn getenv_os(&self, v: &str) -> Option<OsString> {
```

Comment:
I think this might be able to be `&mut self`? If so then the mutex can be removed I believe.

Reply:
I had the same thought, but I got a couple conflicting borrow errors when I tried. I could try again and refactor the code to avoid the mutex if you think it's better.

Reply:
Fixed.